python/lsst/meas/algorithms/detection.py

```diff
@@ -697,14 +701,75 @@ def detectFootprints(self, exposure, doSmooth=True, sigma=None, clearMask=True,
                 self.applyTempLocalBackground(exposure, middle, results)
             self.finalizeFootprints(maskedImage.mask, results, sigma)
 
+            # Compute the significance of peaks after the peaks have been
+            # finalized and local background correction/updatePeaks, so that
+            # the significance represents the "final" detection S/N.
+            results.positive = self.setPeakSignificance(middle, results.positive, results.positiveThreshold)
+            results.negative = self.setPeakSignificance(middle, results.negative, results.negativeThreshold,
+                                                        negative=True)
+
             if self.config.reEstimateBackground:
                 self.reEstimateBackground(maskedImage, results.background)
 
             self.clearUnwantedResults(maskedImage.getMask(), results)
+
             self.display(exposure, results, middle)
 
         return results
 
+    def setPeakSignificance(self, exposure, footprints, threshold, negative=False):
+        """Set the significance of each detected peak to the pixel value divided
+        by the appropriate standard-deviation for ``config.thresholdType``.
+
+        Only sets significance for "stdev" and "pixel_stdev" thresholdTypes.
```

Comment:
I feel like I should know the answer given our previous conversation, but why?

(And a reader other than me would probably have the same question.)

Reply:
I'm not sure how to define a significance value for the `variance` thresholdType (and I don't think we use that anywhere); would it be the ratio of the variance, or of the stddev?

Similar question for `thresholdType=value`: if we just took `peak['peakValue'] / thresholdValue`, that would mean a detection has significance >=1, which could be confusing.

I figured I'd err on the side of doing the calculation for the obvious cases where there was a clear meaning, and leaving the others that I don't think we actually use anywhere.

Reply:
That all sounds reasonable - just put something like that (even just a very brief summary) in the docs.